internal/conf/load.go

```diff
@@ -0,0 +1,243 @@
+// INTECH Process Automation Ltd.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package conf
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"github.com/mitchellh/mapstructure"
+	"gopkg.in/yaml.v3"
+	"io/ioutil"
+	"os"
+	"path"
+	"path/filepath"
+	"strconv"
+	"strings"
+)
+
+const Separator = "__"
+const Prefix = "KUIPER"
+
+func LoadConfig(c interface{}) error {
+	return LoadConfigByName(ConfFileName, c)
+}
+
+func LoadConfigByName(name string, c interface{}) error {
+	dir, err := GetConfLoc()
+	if err != nil {
+		return err
+	}
+	p := path.Join(dir, name)
+	return LoadConfigFromPath(p, c)
+}
+
+func LoadConfigFromPath(path string, c interface{}) error {
+	b, err := ioutil.ReadFile(path)
+	if err != nil {
+		return err
+	}
+	configMap := make(map[string]interface{})
+	err = yaml.Unmarshal(b, &configMap)
+	if err != nil {
+		return err
+	}
+	configs := normalize(configMap)
+	if _, success := c.(*map[string]interface{}); success {
+		names, err := extractKeysFromJsonIfExists(path)
+		if err != nil {
+			return err
+		}
+		applyKeys(configs, names)
+	}
+	err = process(configs, os.Environ())
+	if err != nil {
+		return err
+	}
+	return mapstructure.Decode(configs, c)
+}
+
+func process(configMap map[string]interface{}, variables []string) error {
+	for _, e := range variables {
+		if !strings.HasPrefix(e, Prefix) {
```

Comment:
The Prefix should not be a constant. It should be a variable. When parsing a source config, the prefix is the source name. For example, EDGEX__DEFAULT__PROTOCOL.

Reply:
ok so how can I determine prefix ?